code/api/views.py

```diff
@@ -331,57 +332,70 @@ def post(self, request, author_id):
 
                 # check if this is the correct endpoint
                 if followee_id != author_id:
-                    raise ValueError()
+                    raise ValueError("Object ID does not match inbox ID")
                 
                 inbox = get_object_or_404(Inbox, author_id=followee_id)
 
                 # add follow request to inbox
                 try:
-                    followerAuthor = Author.objects.get(id=follower_id)
+                    followerAuthor = LocalAuthor.objects.get(id=follower_id)
                     inbox.follow_requests.add(followerAuthor)
-                except Author.DoesNotExist:
+                except LocalAuthor.DoesNotExist:
                     raise ValueError()                      
 
                 return HttpResponse(status=200)
 
             elif data["type"] == "like":
                 # https://www.youtube.com/watch?v=VoWw1Y5qqt8 - Abhishek Verma
-                # extract data from reqest body
+                # For what was this reference used?
+
+                # extract data from request body
                 splitObject = data["object"].split("/")
-                object = data["object"].split("/")[-2]
-                id = splitObject[-1]
+                object = data["object"].split("/")[-2]  # Feels a bit dangerous, what if there is a trailing '/'?
+                id = splitObject[-1] # Dangerous
+                liking_author_url = data["author"]["id"]
+
+                # let's try to use url_parser.parse_something(data["object"]) for parsing url
```

Comment:
okay, thanks

Reply:
parsing url now 4deeb34b413e19483cde033eca88eff7fc8f7477. 

Thanks a lot for pulling from my branch! I think you already have resolved all merge conflicts. Can you please also keep the URL parser code from here after I merge?

I think @tylerelston added the reference on line 349. Was this the intended place for it?

Reply:
Removed my comment for now. @tylerelston if you could update the reference with some added context and make sure it's on the correct line on your next PR that would be great.